src/main/scala/zio/keeper/Message.scala

```diff
@@ -1,3 +1,86 @@
 package zio.keeper
 
-final case class Message[S, A](sender: NodeId, state: S, payload: A)
+import java.util.UUID
+
+import zio.keeper.Error.ProtocolDef.SerializationError
+import zio.nio.channels.AsynchronousSocketChannel
+import zio.{ Chunk, IO, ZIO }
+
+import scala.reflect.ClassTag
+
+final case class Message[A](header: Header, payload: A, replyTo: AsynchronousSocketChannel)
+
+case class Header(
```

Comment:
I'd keep `Header` in `Message`'s companion.

Reply:
done